rest/server.go

```diff
@@ -183,6 +185,12 @@ func (server *HTTPServer) WithCORS(cors *CORSOptions) *HTTPServer {
 	return server
 }
 
+// WithGracefulShutdownTimeoutSeconds sets the connection draining timeouts on server shutdown
+func (server *HTTPServer) WithGracefulShutdownTimeoutSeconds(timeout uint32)  *HTTPServer {
```

Comment:
NIT: How about the name is `WithShutdownTimeout`. In the comment for this method we can have the units so that the caller knows that it is in seconds.

Reply:
also please add unit test.